python/lsst/meas/modelfit/display/cModelDisplay.py

```diff
@@ -0,0 +1,264 @@
+                               footBBox)
+
+    # Build the psf
+    shapeletPsfKey = shapelet.MultiShapeletFunctionKey(
+        record.schema["modelfit_DoubleShapeletPsfApprox"])
+    psfApprox = record.get(shapeletPsfKey)
+
+    # Build the dev Image from the shapelet function
+    devIm = afwImage.ImageD(footBBox)
+    dev = dev.convolve(psfApprox)
+    dev.evaluate().addToImage(devIm)
+
+    # Build the exp image from the shapelet function
+    expIm = afwImage.ImageD(footBBox)
+    exp = exp.convolve(psfApprox)
+    exp.evaluate().addToImage(expIm)
+
+    # Build the joint image from the shapelet function
+    jointIm = afwImage.ImageD(footBBox)
+    jointDev = jointDev.convolve(psfApprox)
+    jointExp = jointExp.convolve(psfApprox)
+    jointDev.evaluate().addToImage(jointIm)
+    jointExp.evaluate().addToImage(jointIm)
+
+    # Scale each image by the reference flux
+    devImArr = devIm.getArray()
+    devImArr *= refFlux
+    expImArr = expIm.getArray()
+    expImArr *= refFlux
+    jointImArr = jointIm.getArray()
+    jointImArr *= refFlux
+    return subImage, devIm, expIm, jointIm
+
+
+def reconstructCModel(exposure, record, cModelConfig):
+    """ Reconstruct the CModel for the given record
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure object that contains the source which was  modeled
+    record : `lsst.afw.table.SourceRecord`
+        Record object which contains the measurements made on the source
+    cModelConfig : `lsst.meas.modelfit.CModel(SingleFrame/Forced)Config`
+        Configuration object of the CModel plugin used in the measurement
+        process
+
+    Returns
+    -------
+    devShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model of the dev component of CModel
+    expShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model fo the exp component of CModel
+    devJointShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model of the dev component of CModel jointly
+        fit with the exp component
+    expJointShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model of the exp component of Cmodel jointly
+        fit with the dec component
+    fitSysToMeasSys.flux : 'double'
+        Reference flux used to construct coordinate transformation between
+        the pixel flux system and the model flux system
+    """
+
+    # build a unit system transformation object
+    center = record.getCentroid()
+    position = exposure.getWcs().pixelToSky(center)
+    measSys = measMod.UnitSystem(exposure)
+    approxFlux = record.get("base_PsfFlux_flux")
+    fitSys = measMod.UnitSystem(position, exposure.getCalib(), approxFlux)
+    fitSysToMeasSys = measMod.LocalUnitTransform(position, fitSys, measSys)
+
+    # Build the Shapelet objects
+    ctrl = cModelConfig.makeControl()
+    baseKey = "modelfit_CModel"
```

Comment:
Probably best not to use `Key` as a suffix for a string variable that's used to look up values in records - too easy to confuse with true `Key`s (e.g. I could see someone looking at some of the code below in the future being confused why `Key` doesn't have a `.format` method).